### PR TITLE
Fluentd cleanup

### PR DESCRIPTION
**Pull request by jrcamp**
Gets rid of splunk hec token, ingestHost, ingestPort, ingestProtocol. Most users will be using splunkRealm option instead.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,17 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
...
+### Changed
+
+- fluentd logs are now sent through the collector instead of being sent directly to the backend.
+
+### Removed
+
+- Removed `ingestHost`, `ingestPort`, `ingestProtocol`, use `ingestUrl` instead.
+- Removed `logsBackend`, configure `splunk_hec` exporter directly.
+- Removed `splunk.com/index` annotation for logs.
+- Removed `fluentd.config.indexFields` as all fields sent are indexed.
```
Comment (dmitryax) [inline CHANGELOG.md:16]: The changelog linter doesn't like the dots at the end 😂

**Review by dmitryax — APPROVED** [review]
LGTM

File: CHANGELOG.md [hunk 2]
```diff
@@ -4,6 +4,17 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
 
 ## [Unreleased]
 
+### Changed
+
+- fluentd logs are now sent through the collector instead of being sent directly to the backend.
```
Comment (dmitryax) [inline CHANGELOG.md:9]: Can you add the PR number at the end (#109) ?

File: helm-charts/splunk-otel-collector/templates/daemonset.yaml [hunk 3]
```diff
@@ -63,6 +63,9 @@ spec:
             valueFrom:
               fieldRef:
                 fieldPath: metadata.name
+          {{- with .Values.fluentd.extraEnvs }}
```
Comment (jrcamp) [inline helm-charts/splunk-otel-collector/templates/daemonset.yaml:66]: @dmitryax added back, thanks for catching :)

**Pull request merged**